lib/widgets/post.dart

```diff
@@ -419,12 +422,63 @@ class PostWidget extends HookWidget {
               postImage()
             else if (post.post.url != null && post.post.url.isNotEmpty)
               linkPreview(),
-            if (post.post.body != null)
-              // TODO: trim content
+            if (post.post.body != null && fullPost)
               Padding(
                   padding: const EdgeInsets.all(10),
                   child:
                       MarkdownText(post.post.body, instanceHost: instanceHost)),
+            if (post.post.body != null && !fullPost)
+              LayoutBuilder(
+                builder: (context, constraints) {
+                  final span = TextSpan(
+                    text: post.post.body,
+                  );
+                  final tp = TextPainter(
+                    text: span,
+                    maxLines: 20,
+                    textDirection: Directionality.of(context),
+                  )..layout(maxWidth: constraints.maxWidth - 20);
+
+                  if (tp.didExceedMaxLines) {
+                    return Container(
```

Comment:
Use `ConstrainedBox` instead of `Container`, it focuses on only one thing making it more performant